### ISSUE TITLE
Newsletter subscription does not set the correct store_id if already subscribed.

### ISSUE THREAD

**Issue by bbmxxc** [magento/magento2#19131]
### Preconditions (*)

1. latest Magento Version 2.3-dev
2. Have at least two stores ( A and B)
### Steps to reproduce (*)

1. Register a user in  A
2. Subscribe to Newsletter in A 
3. Switch to Store B
4. Subscribe to Newsletter again

### Expected result (*)

1. When in store B, Subscriber::loadByEmail() should work
2.` SELECT store_id, customer_id FROM newsletter_subscriber` should look like this:

```
+----------+-------------+
| store_id | customer_id |
+----------+-------------+
|        1 |           1 |
|        2 |           1 |
```
 

### Actual result (*)

1. When in store B, loadByEmail() (or loadById() ) cant find the subscriber since
2. `SELECT store_id, customer_id FROM newsletter_subscriber` looks like this:
```
+----------+-------------+
| store_id | customer_id |
+----------+-------------+
|        1 |           1 |
|        1 |           1 |

```

### What i think where the error is
In `app/code/Magento/Newsletter/Model/Subscriber.php` starting on line [468](https://github.com/magento/magento2/blob/6481cecaa9ea6959ebb255b4271f88d746ebb21b/app/code/Magento/Newsletter/Model/Subscriber.php#L468): 
If the subscriber already exists with this email, the new entry gets the `$customer->getStoreId()`
not the `$this->_storeManager->getStore()->getId()`

If this is the intended behavior then `loadByEmail()` or `loadByCustomerId()` should be store_id agnostic.

Cheers!
Ben

**Comment by magento-engcom-team** [comment]
Hi @bbmxxc. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@bbmxxc do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @bbmxxc thank you for you report this issue has already fixed in 2.3-dev branch
![selection_198](https://user-images.githubusercontent.com/40756439/48266239-7a3dba00-e437-11e8-91f7-11b770b710f0.png)
